homeassistant/components/broadlink/switch.py

```diff
@@ -32,13 +35,24 @@
     }
 )
 
+OLD_SWITCH_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_COMMAND_OFF): data_packet,
+        vol.Optional(CONF_COMMAND_ON): data_packet,
+        vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+    }
+)
+
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_HOST): cv.string,
         vol.Optional(CONF_MAC): mac_address,
-        vol.Optional(CONF_SWITCHES, default=[]): vol.All(
-            [SWITCH_SCHEMA], cv.ensure_list
+        vol.Optional(CONF_SWITCHES, default=[]): vol.Any(
+            cv.schema_with_slug_keys(OLD_SWITCH_SCHEMA),
+            vol.All([SWITCH_SCHEMA], cv.ensure_list),
```

Comment:
The order is important for that validator. We need to ensure a list before passing the value to the list schema.
```suggestion
            vol.All(cv.ensure_list, [SWITCH_SCHEMA]),
```